Geometry/HcalCommonData/src/HcalDDDRecConstants.cc

```diff
@@ -185,7 +185,7 @@ HcalDDDRecConstants::getHCID(int subdet, int keta, int iphi, int lay,
     ++phi0;
     unit     = hcons.unitPhi(phibin[eta-1]);
     phi      = hcons.phiNumber(phi0,unit);
-    depth    = hcons.findDepth(subdet,eta,phi,zside,lay);
+    depth    = hcons.findDepth(subdet,eta,phi,zside,lay-1);
```

Comment:
Hi @bsunanda -so to summarize the issue, HcalDDDRecConstants.cc expects layers starting from "1" and HcalDDDSimConstants expects layers starting from "0"?